netbox/extras/scripts.py

```diff
@@ -446,6 +511,41 @@ def load_json(self, filename):
 
         return data
 
+    def run_tests(self):
+        """
+        Run the report and save its results. Each test method will be executed in order.
+        """
+        self.logger.info(f"Running report")
+
+        try:
+            for method_name in self.test_methods:
+                self._current_method = method_name
+                test_method = getattr(self, method_name)
+                test_method()
+        except Exception as e:
+            self.post_run()
+            self._current_method = ''
+            raise e
+
+        self._current_method = ''
+
+    def run(self, data, commit):
+        self.pre_run()
+        self.run_tests()
+        self.post_run()
```

Comment:
We should move `pre_run()` and `post_run()` to the `run_script()` function, to ensure that they are always called if `run()` is overridden.

Reply:
On second thought, let's leave these as-is. They should be considered legacy report functions and ultimately will be phased out.